esrally/mechanic/provisioner.py

```diff
@@ -184,7 +186,17 @@ def _provisioner_variables(self):
         plugin_variables = {}
         mandatory_plugins = []
         for installer in self.plugin_installers:
-            mandatory_plugins.append(installer.plugin_name)
+            # For Elasticsearch < 6.3 more specific plugin names are required for mandatory plugin check
+            # Details in: https://github.com/elastic/elasticsearch/pull/28710
+            try:
+                if ((versions.major_version(self.distribution_version) == 6 and
+                     versions.minor_version(self.distribution_version) < 3) or
+                    (versions.major_version(self.distribution_version) < 6)):
+                    mandatory_plugins.append(installer.sub_plugin_name)
+                else:
+                    raise NameError
+            except (NameError, TypeError, exceptions.InvalidSyntax):
```

Comment:
I *think* a `TypeError` should not occur at this point? If the individual components of the version string are not `int` they would not match the regex.

Reply:
Yup it was added because the integration tests raise this, if not present:

```
[ERROR] Cannot race. expected string or bytes-like object
	Traceback (most recent call last):
  File "/home/dl/source/elastic/dliappis_forks/rally/.tox/py36/lib/python3.6/site-packages/esrally/mechanic/mechanic.py", line 516, in receiveMsg_StartNodes
    nodes = self.mechanic.start_engine()
  File "/home/dl/source/elastic/dliappis_forks/rally/.tox/py36/lib/python3.6/site-packages/esrally/mechanic/mechanic.py", line 645, in start_engine
    node_configs.append(p.prepare(binaries))
  File "/home/dl/source/elastic/dliappis_forks/rally/.tox/py36/lib/python3.6/site-packages/esrally/mechanic/provisioner.py", line 164, in prepare
    provisioner_vars = self._provisioner_variables()
  File "/home/dl/source/elastic/dliappis_forks/rally/.tox/py36/lib/python3.6/site-packages/esrally/mechanic/provisioner.py", line 192, in _provisioner_variables
    if ((versions.major_version(self.distribution_version) == 6 and
  File "/home/dl/source/elastic/dliappis_forks/rally/.tox/py36/lib/python3.6/site-packages/esrally/utils/versions.py", line 25, in major_version
    major, _, _, _ = components(version)
  File "/home/dl/source/elastic/dliappis_forks/rally/.tox/py36/lib/python3.6/site-packages/esrally/utils/versions.py", line 49, in components
    matches = versions_pattern.match(version)
TypeError: expected string or bytes-like object

```